pkg/network/tracer/offsetguess/tracer.go

```diff
@@ -698,6 +703,22 @@ func (t *tracerOffsetGuesser) flowi6EntryState() GuessWhat {
 	return GuessSAddrFl6
 }
 
+func setupLoopbackInNS(ns netns.NsHandle) error {
+	if err := netns.Set(ns); err != nil {
+		return err
+	}
+	loopback, err := netlink.LinkByName("lo")
```

Comment:
Using the global functions may not guarantee that this request is executed in the right namespace. You should explicitly create a new handle with `NewHandleAt` for the correct namespace and use that here.